packages/devops/scripts/lambda/actions/spin_up_lesmis_deployment.py

```diff
@@ -0,0 +1,71 @@
+# Creates a new deployment of LESMIS with a specific branch checked out
+#
+# Example config
+#
+# 1. Spin up a new feature branch deployment of LESMIS, that will be deleted after 8 hours
+# {
+#   "Action": "spin_up_lesmis_deployment",
+#   "User": "edwin",
+#   "Branch": "wai-965",
+#   "DbSnapshot": "lesmis-on-premise/lesmis-dump-tupaia-202404091048.sql",
+#   "HoursOfLife": 8
+# }
+
+from pathlib import Path
+
+from helpers.create_from_image import create_server_instance_from_image
+from helpers.utilities import find_instances, build_extra_tags
+
+def spin_up_lesmis_deployment(event):
+    # validate input config
+    if 'DeploymentName' not in event:
+        raise Exception('You must include the key "DeploymentName" in the lambda config, e.g. "dev".')
+    deployment_name = event['DeploymentName']
+    branch = event.get('Branch', deployment_name) # branch defaults to deployment name if not specified
+    if deployment_name == 'production' and branch != 'master':
+        raise Exception('The production deployment needs to check out master, not ' + branch)
+
+    # find current instances
+    existing_instances = find_instances([
+        { 'Name': 'tag:DeploymentName', 'Values': [deployment_name] },
+        { 'Name': 'tag:DeploymentType', 'Values': ['lesmis'] },
+        { 'Name': 'instance-state-name', 'Values': ['running', 'stopped'] } # ignore terminated instances
+    ])
+
+    if len(existing_instances) != 0:
+      raise Exception('A deployment already exists, perhaps you want to redeploy and swap out the existing one? The easiest way is to push a new commit.')
+    
+    if 'DbSnapshot' not in event:
+        raise Exception('You must include the file location of a snapshot of the lesmis database in S3')
```

Comment:
Oh nice we're working with S3 snapshots!!
That's awesome, opens a smoother pass for a shared RDS instances with multiple DBs on it (one per dev)

Reply:
Maybe I should rename this? Strictly speaking it's a database dump rather than a snapshot (ie. RDS snapshot). It's just a zipped file of a pg_dump of the LESMIS production database.

I think has to be that way since we don't use RDS with type of deployment.

Reply:
I agree tbh!